src/aiodynamo/credentials.py

```diff
@@ -289,6 +293,81 @@ def and_then(thing: Optional[T], mapper: Callable[[T], U]) -> Optional[U]:
     return thing
 
 
+# https://docs.aws.amazon.com/STS/latest/APIReference/API_AssumeRoleWithWebIdentity.html
+@dataclass
+class AssumeRoleWithWebIdentityCredentials(MetadataCredentials):
+    ACTION = "AssumeRoleWithWebIdentity"
+    VERSION = "2011-06-15"
+
+    timeout: Timeout = 2
+    max_attempts: int = 3
+
+    base_url: URL = URL("https://sts.amazonaws.com/")
+
+    async def fetch_metadata(self, http: HttpImplementation) -> Metadata:
+        config: dict = self._load_config()
+
+        url = self._build_url(**config)
+        response: bytes = await fetch_with_retry_and_timeout(
+            http=http,
+            max_attempts=self.max_attempts,
+            timeout=self.timeout,
+            request=Request(
+                method="GET", url=url, headers=None, body=None
+            ),
+        )
+        credentials = self._parse_aws_response(response)["Credentials"]
+        return Metadata(
+            key=Key(
+                id=credentials["AccessKeyId"],
+                secret=credentials["SecretAccessKey"],
+                token=credentials.get("Token", None),
+            ),
+            expires=parse_amazon_timestamp(credentials["Expiration"]),
+        )
+
+    def is_disabled(self) -> bool:
+        return False
```

Comment:
looking at the implementation of `_load_config`, this should probably check if a `role` (and any other potentially necessary bits) has been provided. Since this loader is added to the default credentials loader list, it's important that `is_disabled` is a bit conservative so that the default loader can skip this if it's not configured.